dotcom-rendering/src/lib/getLiveblogAdPositions.ts

```diff
@@ -0,0 +1,63 @@
+import {
+	calculateApproximateBlockHeight,
+	shouldDisplayAd,
+} from './liveblogAdSlots';
+
+const getLiveblogAdPositions = (
+	blocks: Block[],
+): { mobileAdPositions: number[]; desktopAdPositions: number[] } => ({
+	mobileAdPositions: getAdPositionsForScreenSize({
+		blocks,
+		isMobile: true,
+	}),
+	desktopAdPositions: getAdPositionsForScreenSize({
+		blocks,
+		isMobile: false,
+	}),
+});
+
+/**
+ * Decides where ads should be inserted on liveblogs/deadblogs.
+ */
+const getAdPositionsForScreenSize = ({
+	blocks,
+	isMobile,
+}: {
+	blocks: Block[];
+	isMobile: boolean;
+}): number[] =>
+	blocks.reduce<{
+		heightSinceAd: number;
+		adPositions: number[];
+	}>(
+		(accumulator, block, index) => {
+			const { heightSinceAd, adPositions } = accumulator;
```

Comment:
One last nit I promise!

```suggestion
		({ heightSinceAd, adPositions }, block, index) => {
```

Reply:
Sorry, I missed this